packages/jellyfish-api-core/__tests__/category/governance/listGovProposals.test.ts

```diff
@@ -108,6 +120,234 @@ describe('Governance', () => {
+      type: ListProposalsType.CFP,
+      cycle: 2
+    })
+    expect(proposals.length).toStrictEqual(1)
+    expect(proposals.filter(proposal => proposal.type === ProposalType.COMMUNITY_FUND_PROPOSAL).length).toStrictEqual(1)
+    expect(proposals.filter(proposal => proposal.status === ProposalStatus.VOTING).length).toStrictEqual(1)
+  })
+
+  it('should listGovProposals with specific cycle, status and type', async () => {
+    const proposals = await testing.rpc.governance.listGovProposals({
+      status: ListProposalsStatus.VOTING,
+      type: ListProposalsType.CFP,
+      cycle: 2
+    })
+    expect(proposals.length).toStrictEqual(1)
+    expect(proposals.filter(proposal => proposal.type === ProposalType.COMMUNITY_FUND_PROPOSAL).length).toStrictEqual(1)
+    expect(proposals.filter(proposal => proposal.status === ProposalStatus.VOTING).length).toStrictEqual(1)
+  })
+
+  // Pagination related tests
+  it('should return all result with limit: 0 in listGovProposals', async () => {
+    const proposals = await testing.rpc.governance.listGovProposals({
+      pagination: {
+        limit: 0
+      }
+    })
+    expect(proposals.length).toStrictEqual(5)
+    expect(proposals.filter(proposal => proposal.type === ProposalType.COMMUNITY_FUND_PROPOSAL).length).toStrictEqual(2)
+    expect(proposals.filter(proposal => proposal.type === ProposalType.VOTE_OF_CONFIDENCE).length).toStrictEqual(3)
+    expect(proposals.filter(proposal => proposal.status === ProposalStatus.VOTING).length).toStrictEqual(4)
+    expect(proposals.filter(proposal => proposal.status === ProposalStatus.REJECTED).length).toStrictEqual(1)
+  })
+
+  it('should listGovProposals with specific limit', async () => {
+    const allProposals = await testing.rpc.governance.listGovProposals()
+    const proposals = await testing.rpc.governance.listGovProposals({
+      pagination: {
+        limit: 2
+      }
+    })
+    expect(proposals.length).toStrictEqual(2)
+    expect(proposals).toStrictEqual(allProposals.slice(0, 2))
+  })
+
+  it('should listGovProposals with including_start: false', async () => {
+    const allProposals = await testing.rpc.governance.listGovProposals()
+    const proposals = await testing.rpc.governance.listGovProposals({
+      pagination: {
+        including_start: false
+      }
+    })
+    expect(proposals).toStrictEqual(allProposals.slice(1))
+    expect(proposals.length).toStrictEqual(4)
+  })
+
+  it('should listGovProposals with including_start and limit', async () => {
+    const allProposals = await testing.rpc.governance.listGovProposals()
+    const proposals = await testing.rpc.governance.listGovProposals({
+      pagination: {
+        including_start: false,
+        limit: 2
+      }
+    })
+    expect(proposals).toStrictEqual(allProposals.slice(1, 3))
+    expect(proposals.length).toStrictEqual(2)
+  })
+
+  // TODO: remove `skip` after blockchain fixes bug where `start` is not considered when non-all status is not passed
```

Comment:
just wondering if there is an existing PR or issue for this bug on ain?

Reply:
It's being worked on ATM, but no PR yet.

Reply:
PR created yesterday https://github.com/DeFiCh/ain/pull/1689